### DISCUSSION TITLE
Klipper Not Upgradeable Through Mainsail Update Manager

### DISCUSSION THREAD

**Discussion by Rulevoid** [Guilouz/Creality-Helper-Script-Wiki#442] [Issues]
I am using an Ender 3 V3 KE.  On my V2 with a Raspberry Pi and Mainsail OS, I see Klipper and it's upgradeable from the Mainsail Update Manager.  However, on my V3 KE using this K1 install script, I do not see an entry for Klipper in the Update Manager.  How do I update Klipper and how can I add it to the Update Manager?

**Comment by Guilouz** [comment]
Yes it's normal, Klipper can't be updated because Creality use a modified build.

**Reply by Rulevoid** [reply]
Got it. So how do you upgrade to Creality latest Klipper? There's got to still be a way to update that when updates are available, right?

**Reply by Ackta** [reply]
follow

**Reply by Guilouz** [reply]
Since all the firmware updates already released Creality has not updated Klipper, they just belong to changes in some files so you should not hope for an update of Klipper by Creality, it will never happen.

**Reply by B-Man1** [reply]
> Since all the firmware updates already released Creality has not updated Klipper, they just belong to changes in some files so you should not hope for an update of Klipper by Creality, it will never happen.

Is it possible/ should we switch to mainline klipper instead of Creality's version so that Moonraker can update klipper as the OP is looking for?  I'm using a K1 Max with the klipper helper script and was also looking for "klipper" to appear in the update manager in mainsail

**Comment by salami738** [comment]
Klipper is Python code, right? Shouldn't it be possible to pull a diff from the Ender 3v3 KE filesystem and compare it with Mainline? Then you could see if reintegration is possible.

**Reply by salami738** [reply]
See also: https://forum.creality.com/t/how-to-update-klipper-and-get-adaptive-mesh-working-on-ender-3-v3-regular/32599
And: https://github.com/CrealityOfficial/Ender-3_V3_KE_Klipper
